python/ray/dashboard/client/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+type ThemeMode = "light" | "dark";
+
+type ThemeContextType = {
+  mode: ThemeMode;
+  toggleTheme: () => void;
+};
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+type ThemeProviderProps = {
+  children: ReactNode;
+};
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
+  const [mode, setMode] = useState<ThemeMode>(() => {
+    // Get initial theme from localStorage or default to light
+    try {
+      const savedTheme = localStorage.getItem(
+        "ray-dashboard-theme",
+      ) as ThemeMode;
```

Comment:
instead of manually calling localStorage with useEffects, let's use the `useLocalStorage` hook from the `usehooks-ts` library.